app/javascript/controllers/work_type_controller.js

```diff
@@ -23,17 +24,59 @@ export default class extends Controller {
     this.areaTarget.hidden = false
   }
 
+  toggleMoreTypes(event) {
+    event.preventDefault()
+
+    this.moreTypesTarget.hidden ?
+      this.showMoreTypes() :
+      this.hideMoreTypes()
+  }
+
+  showMoreTypes() {
+    this.moreTypesTarget.hidden = false
+    this.moreTypesLinkTarget.innerHTML = 'See fewer options'
+    this.moreTypesLinkTarget.classList.toggle('collapsed', false)
+    this.continueButtonTarget.focus()
+  }
+
+  hideMoreTypes() {
+    this.moreTypesTarget.hidden = true
+    this.moreTypesLinkTarget.innerHTML = 'See more options'
+    this.moreTypesLinkTarget.classList.toggle('collapsed', true)
+  }
+
   displaySubtypeOptions(type) {
-    const subtypes = document.subtypes[type].map((subtype)=> {
-      const id = subtype.replace(/\s/g, '_')
-      return this.templateTarget.innerHTML.replace(/SUBTYPE_LABEL/g, subtype).replace(/SUBTYPE_ID/g, id)
-    })
-    this.subtypeTarget.innerHTML = subtypes.join('')
+    // Show the more options link
+    this.moreTypesLinkTarget.hidden = false
+    this.subtypeTarget.innerHTML = this.subtypesFor(type).join('')
+    this.moreTypesTarget.innerHTML = this.moreTypesFor(type).join('')
     this.areaTarget.innerHTML = this.templateHeaderTarget.innerHTML
   }
 
   displayOtherSubtypeOptions() {
+    // Hide the more options link
+    this.moreTypesLinkTarget.hidden = true
+    this.hideMoreTypes()
     this.subtypeTarget.innerHTML = this.otherTemplateTarget.innerHTML
     this.areaTarget.innerHTML = this.otherTemplateHeaderTarget.innerHTML
   }
+
+  subtypesFor(type) {
+    return document
+      .subtypes[type]
+      .map((subtype)=> {
+        const id = subtype.replace(/\s/g, '_')
+        return this.templateTarget.innerHTML.replace(/SUBTYPE_LABEL/g, subtype).replace(/SUBTYPE_ID/g, id)
+      })
+  }
+
+  moreTypesFor(type) {
+    return document
+      .moreTypes
+      .filter(moreType => !document.subtypes[type].includes(moreType))
```

Comment:
The "more" types for a work type should not include its own subtypes.

Reply:
Can you add a comment to this effect. It would help to provide a specific example of this.

Reply:
@jcoyne You bet. See above!